pkg/router/template/router.go

```diff
@@ -301,6 +302,17 @@ func genCertificateHostName(hostname string, wildcard bool) string {
 	return hostname
 }
 
+func randomEndpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit, random bool) []Endpoint {
+	endpoints := endpointsForAlias(alias, svc)
+	if random {
+		for i := range endpoints {
+			rIndex := rand.Intn(i + 1)
```

Comment:
Reading https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle, this isn't quite right... we need to either go backwards, or get an integer such that i ≤ j < n.  It's easier to go backwards:

```
for i := len(slice) - 1; i > 0; i-- {
    j := rand.Intn(i + 1)
    slice[i], slice[j] = slice[j], slice[i]
}
```